types/react-select/lib/Creatable.d.ts

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import Select, { Props as SelectProps } from './Select';
+import { OptionsType, ValueType, ActionMeta } from './types';
+import { cleanValue } from './utils';
+import manageState from './stateManager';
+
+export interface CreatableProps<OptionType> {
+  /* Allow options to be created while the `isLoading` prop is true. Useful to
+     prevent the "create new ..." option being displayed while async results are
+     still being loaded. */
+  allowCreateWhileLoading?: boolean;
+  /* Gets the label for the "create new ..." option in the menu. Is given the
+     current input value. */
+  formatCreateLabel?: (inputValue: string) => Node;
+  /* Determines whether the "create new ..." option should be displayed based on
+     the current input value, select value and options array. */
+  isValidNewOption?: (a: string, b: ValueType<OptionType>, c: OptionsType<OptionType>) => boolean;
```

Comment:
It seems that, at runtime, the second argument is only ever an array with at least a single option inside it.

If it wasn't the case it'd crash at runtime in its default implementation: https://github.com/JedWatson/react-select/blob/b34589446345a6d90e353a97ab7eff1700423e33/src/Creatable.js#L55